### PR TITLE
fix: issue where wagmi disconnect hook would not update appkit state

### PR DESCRIPTION
**Pull request by tomiir**
# Description
- Fix issue where wagmi's disconnect hook would not update internal state

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

Closes APKT-1186

# Checklist

- [X] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [X] My changes generate no new warnings
- [X] I have reviewed my own code
- [X] I have filled out all required sections
- [ ] I have tested my changes on the preview link
- [ ] Approver of this PR confirms that the changes are tested on the preview link

### CODE REVIEW

File: packages/adapters/wagmi/src/client.ts [hunk 1]
```diff
@@ -600,14 +600,16 @@ export class WagmiAdapter implements ChainAdapter {
       | 'status'
     >
   >) {
-    const isConnected = ChainController.state.activeCaipAddress
-
-    if (status === 'disconnected' && !isConnected) {
+    const isAuthConnector = connector?.id === ConstantsUtil.AUTH_CONNECTOR_ID
```
Comment (zoruka) [inline packages/adapters/wagmi/src/client.ts:603]: should we move the constant to just before where it is used?

**Review by zoruka — COMMENTED** [review]
would it be possible to create a e2e test to run over the new test component?

**Review by zoruka — APPROVED** [review]

**Comment by LucasALLOIN** [comment]
Hey team, can we have an ETA on when this PR will be merge? This is really blocking on our side.

File: apps/laboratory/src/components/Wagmi/WagmiTests.tsx [hunk 2]
```diff
@@ -70,6 +71,13 @@ export function WagmiTests() {
             </Heading>
             <WagmiSendCallsWithPaymasterServiceTest />
           </Box>
+
+          <Box>
+            <Heading size="xs" textTransform="uppercase" pb="2">
```
Comment (enesozturk) [inline apps/laboratory/src/components/Wagmi/WagmiTests.tsx:76]: Maybe we should add this case to the `wallet.spec.ts` and `email.spec.ts` just like we do for `w3m-open-hook-button` (custom hook button)

**Review by enesozturk — APPROVED** [review]

File: apps/laboratory/src/components/Wagmi/WagmiTests.tsx (hunk 2 above)
Comment (tomiir) [reply apps/laboratory/src/components/Wagmi/WagmiTests.tsx:76]: we should add a disconnect hook ourselves. Will do on a different PR cause `appkit.disconnect()` is currently not working 

For ethers there's no equivalent

**Pull request merged**